cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -155,7 +157,14 @@ func BuildAzureConfig(configReader io.Reader) (*Config, error) {
 		cfg.Location = os.Getenv("LOCATION")
 		cfg.ResourceGroup = os.Getenv("ARM_RESOURCE_GROUP")
 		cfg.TenantID = os.Getenv("ARM_TENANT_ID")
+		if tenantId := os.Getenv("AZURE_TENANT_ID"); tenantId != "" {
+			cfg.TenantID = tenantId
+		}
 		cfg.AADClientID = os.Getenv("ARM_CLIENT_ID")
+		if clientId := os.Getenv("AZURE_CLIENT_ID"); clientId != "" {
+			cfg.AADClientID = clientId
+		}
```

Comment:
why doing an override here?

Reply:
These override are here because of the project https://github.com/Azure/azure-workload-identity,
The default environment values it sets are:
- AZURE_TENANT_ID
- AZURE_CLIENT_ID
- AZURE_FEDERATED_TOKEN_FILE
- AZURE_AUTHORITY_HOST

To have better interoperability I added these aliases. I can remove them though and add them in a new PR

Reply:
Thanks for sharing the contexts, it makes sense.

Reply:
@feiskyer thank you for the feedback!